src/utils/colors/index.js

```diff
@@ -0,0 +1,51 @@
+// https://chir.ag/projects/name-that-color/
+export const colors = {
+  bondi_blue: '#00ADB5',
+  white: '#ffffff',
+  black: '#000000',
+  emperor: '#4F4F4F',
+  gray: '#828282',
+  gray1: '#C4C4C4',
+  lightgrey: '#F5F5F5',
+  concrete: '#F2F2F2',
+  mine_shaft: '#333333',
+  bunting: '#11243D',
+  alto: '#E0E0E0',
+  porcelain: '#ecf0f1',
+  silver: '#BDBDBD',
+  french_pass: '#b6e4fd',
+  pattens_blue: '#ddf2fe',
+  blue: '#2F80ED',
+  lightblue: '#55C2FF',
+  blue1: '#2C67BC',
+  red: '#FF2E63',
+  holytosca: '#00ADB5',
+  holytosca30percent: '#00ADB54D',
+  blackgrey: '#828282',
+  redalert: '#FF2E63',
+  halfBaked: '#88CDD0',
+  tradewind: '#58B1B5',
+  elm: '#22878B',
+  blockColor: '#FF2E63',
+  darkBlue: '#2C67BC',
+  blueSea10: '#55C2FF',
+  greenDark: '#004346',
+  babyBlue: '#7B9DCD',
+  bluePrimary: '#4782D7',
+  greySubtile: '#69707C',
+  greySubtile1: '#9B9FA9',
+  greySubtile2: '#E8EBED',
+};
+
+export const hexToRgb = (hex, opacity = 1) => {
+  const result = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(hex);
+  const r = parseInt(result[1], 16);
+  const g = parseInt(result[2], 16);
+  const b = parseInt(result[3], 16);
+  return result ? `rgba(${r},${g},${b},${opacity})` : null;
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `hexToRgb` function does not handle invalid hex color inputs gracefully. If the regex match fails, it will still try to parse undefined values, which can lead to a runtime error. Consider adding an early return if the regex match fails.

```diff
export const hexToRgb = (hex, opacity = 1) => {
  const result = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(hex);
+   if (!result) return null;
  const r = parseInt(result[1], 16);
  const g = parseInt(result[2], 16);
  const b = parseInt(result[3], 16);
  return `rgba(${r},${g},${b},${opacity})`;
};
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->